runner/engine_test.go

```diff
@@ -1058,3 +1058,86 @@ include_file = ["main.sh"]
 	}
 	assert.Equal(t, []byte("modified"), bytes)
 }
+
+type testExiter struct {
+	t          *testing.T
+	called     bool
+	expectCode int
+}
+
+func (te *testExiter) Exit(code int) {
+	te.called = true
+	if code != te.expectCode {
+		te.t.Fatalf("expected exit code %d, got %d", te.expectCode, code)
+	}
+}
+
+func TestEngineExit(t *testing.T) {
```

Comment:
I'm unsure I get it

You ate no longer calling stopBin, so here you are testing the test exiter behavior? 

I mean I don't get you are calling the exiter itself in this method

🤔

Did I miss something?

You had the call to stopBin earlier, no?